### ISSUE TITLE
SIGSEGV on WSL 2

### ISSUE THREAD

**Issue by ClementNerma** [DioxusLabs/dioxus#3354]
**Problem**

When compiling a basic example on WSL 2, the program crashes without displaying anything. 

When running it with `strace`, the process crashes with `SIGSEGV` (so a segmentation fault).

I don't really know how to fix that, I could give you a complete `strace` log if that's useful?

**Steps To Reproduce**

1. Create a new project
2. Add `dioxus` with the `desktop` feature
3. Use the following `src/main.rs` file:

```rust
use dioxus::prelude::*;

fn main() {
    dioxus::launch(|| rsx! { "hello dioxus! ­ƒº¼" });
}
```

4. `cargo run`
5. Enjoy the crash

**Environment:**
 - Dioxus version: `0.6.0`
 - Rust version: `1.82.0`
 - OS info: Windows 11 24H2, WSL `2.3.26.0` with Ubuntu 24.10, kernel `5.15.167.4-1`, WSLg `1.0.65`
 - App platform: `desktop`

**Comment by DogeDark** [comment]
Running desktop in WSL has been tricky. Did you take a look at https://dioxuslabs.com/learn/0.6/getting_started/#wsl?

**Comment by ClementNerma** [comment]
Yeah I have the required dependencies installed, the `DISPLAY` env variable set to `:0`, the latest WSL version + Linux kernel (at least the one available in Ubuntu 24.10).

Still doesn't work. What surprises me is not the fact it's not launching (there could be tons of incompatibility issues in terms of libraries interop and dependencies) but the fact that it crashes silently with a segmentation fault.

**Comment by Notliam99** [comment]
Hi i running into the same issue not sure if related due to me running on arch on bare metal

The desktop application fails to run *not build* i get the error on run under [niri](https://github.com/YaLTeR/niri/) wayland desktop enviroment. but under gnome wayland it works fine. 

```bash
Application [linux] exited with error: signal: 11 (SIGSEGV) (core dumped)
```

- dioxus version: `0.6.0`
- rust version: 1.83.0
- os: archlinux on not under wsl ect
- desktop enviroment: [niri](https://github.com/YaLTeR/niri)
- platform: `desktop linux`
